subsys/net/openthread/report/report_generation_utils.cmake

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Usage:
+#   ot_report_git_version(<output> <directory>)
+#
+# This function uses the git describe command to read the current revision
+# of the <directory> directory and saves the result in the <output> variable
+#
+#
+# directory: Directory that contains checked out the git repository of which
+#            you want to read the revision.
+# output: Output variable that will contain the read revision.
+#
+function(ot_report_git_version output directory)
+    execute_process(
+        COMMAND git describe --dirty --always
+        WORKING_DIRECTORY ${directory}
+        OUTPUT_VARIABLE GIT_REV OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_QUIET
+    )
+    set(${output} "${GIT_REV}" PARENT_SCOPE)
+endfunction()
+
+# Usage:
+#   ot_report_add_message(<message> <to_stdout>)
+#
+# This function writes the <message> text to the file which is defined by
+# the CONFIG_OPENTHREAD_REPORT_BUILD_ARTEFACT_NAME kconfig value located in the
+# application build directory.
+#
+# The function will not write anything to the required file if
+# the CONFIG_OPENTHREAD_REPORT_BUILD_ARTEFACT kconfig is not set to y.
+#
+# If the <to_stdout> argument is set to TRUE then the message will be printed
+# to the stdout console.
+#
+#
+# message: Text message to be written
+# to_stdout: Boolean to decide whether the text message should be shown in the
+#            std console.
+#
+function(ot_report_add_message message to_stdout)
+    if(CONFIG_OPENTHREAD_REPORT_BUILD_ARTEFACT)
+        set(ARTEFACT ${CMAKE_BINARY_DIR}/${CONFIG_OPENTHREAD_REPORT_BUILD_ARTEFACT_NAME})
+        file(APPEND ${ARTEFACT} ${message}\n)
+    endif()
+    if(${to_stdout})
+        message(${message})
+    endif()
+endfunction()
+
+# Usage:
+#   ot_report_git_diff(<OUTPUT> <BASE_REVISION> <BASE_DIRECTORY> <SUBDIRECTORY>)
+#
+# This function generates git difference between the <BASE_REVISION> and the current
+# revision that is checked out within the <BASE_DIRECTORY>.
+# You can provide the <SUBDIRECTORY> argument to narrow down the scope of the diff
+# function.
+#
+# For example to compare the current revision of the nrfxlib to the state from
+# NCS 2.7.0 release and take into account only the openthread directory use the following
+# invocation:
+#
+# ot_report_git_diff(MY_VARIABLE v2.7.0 ${ZEPHYR_NRFXLIB_MODULE_DIR} "openthread")
+#
+# OUTPUT: Output variable that contains the git diff.
+# BASE_REVISION: The revision to be compared with the current one.
+# BASE_DIRECTORY: Directory that contains the git repository of which
+#                 you want to read the revision.
+# SUBDIRECTORY: A subdirectory within the BASE_DIRECTORY to be checked by the git
+#               diff command.
+#
+function(ot_report_git_diff OUTPUT BASE_REVISION BASE_DIRECTORY SUBDIRECTORY)
```

Comment:
Could you change argument names to lower case to keep consistency with other functions?